writer.go

```diff
@@ -1291,23 +1287,26 @@ func (c *writerColumn) reset() {
 	c.offsetIndex.PageLocations = c.offsetIndex.PageLocations[:0]
 }
 
-func (c *writerColumn) totalRowCount() int64 {
+func (c *ColumnWriter) totalRowCount() int64 {
 	n := c.numRows
 	if c.columnBuffer != nil {
 		n += int64(c.columnBuffer.Len())
 	}
 	return n
 }
 
-func (c *writerColumn) flush() (err error) {
+func (c *ColumnWriter) flush() (err error) {
+	if c.columnBuffer == nil {
+		return nil
+	}
```

Comment:
This was a possible nil-dereference panic. It looks like this is called when the writer is flushed, so if no rows were ever written, it would panic on the following line.